sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -180,7 +181,11 @@ def _get_request_data(self, request):
         """
         if request.files:
             form_data = aiohttp.FormData(request.data or {})
-            for form_file, data in request.files.items():
+            if isinstance(request.files, Mapping):
+                files = list(request.files.items())
+            else:
+                files = request.files
+            for form_file, data in files:
```

Comment:
we should change it so `files` after request creation is a list of tuples, regardless of whether `data` is present. If `data` is present, we can add it to the list of tuples

Reply:
Updated